compiler/rustc_target/src/callconv/x86_64.rs

```diff
@@ -186,6 +186,11 @@ where
             // Not touching this...
             return;
         }
+        if is_arg && arg.layout.pass_indirectly_in_non_rustic_abis(cx) {
+            int_regs = int_regs.saturating_sub(1);
```

Comment:
Very odd that this would happen here but not for unsized arguments... it's pre-existing, but can you add a FIXME? (Also for the other files with the same pattern)